src/test/java/org/takes/facets/auth/PsAllTest.java

```diff
@@ -0,0 +1,170 @@
+
+import com.jcabi.aspects.Tv;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import org.hamcrest.MatcherAssert;
+import org.hamcrest.Matchers;
+import org.junit.Test;
+import org.takes.Response;
+import org.takes.rq.RqFake;
+import org.takes.rs.RsEmpty;
+
+/**
+ * Test of {@link PsAll}.
+ * @author Georgy Vlasov (wlasowegor@gmail.com)
+ * @version $Id$
+ * @since 0.22
+ */
+public final class PsAllTest {
+
+    /**
+     * Fails if PsAll with 0 Passes is created.
+     * @throws Exception If fails
+     */
+    @Test(expected = IllegalArgumentException.class)
+    public void thereShouldBeAtLeastOnePass() throws Exception {
+        MatcherAssert.assertThat(
+            new PsAll(
+                new ArrayList<Pass>(0),
+                0
+            ).enter(new RqFake()).has(),
+            Matchers.is(false)
+        );
+    }
+
+    /**
+     * Fails if index is greater or equal to the number of Passes to enter.
+     * @throws Exception If fails
+     */
+    @Test(expected = IllegalArgumentException.class)
+    public void indexMustBeSmallEnough() throws Exception {
+        MatcherAssert.assertThat(
+            new PsAll(
+                Arrays.asList(
+                    new PsFake(true),
+                    new PsFake(false)
+                ),
+                2
+            ).enter(new RqFake()).has(),
+            Matchers.is(false)
+        );
+    }
+
+    /**
+     * PsAll with a single Pass that can be entered should succeed.
+     * @throws Exception If fails
+     */
+    @Test
+    public void testOneSuccessfull() throws Exception {
+        MatcherAssert.assertThat(
+            new PsAll(
+                Collections.singletonList(new PsFake(true)),
+                0
+            ).enter(new RqFake()).has(),
+            Matchers.is(true)
+        );
+    }
+
+    /**
+     * Fail with only one failure test.
+     * @throws Exception if exception
+     */
+    @Test
+    public void testOneFail() throws Exception {
+        MatcherAssert.assertThat(
+            new PsAll(
+                Collections.singletonList(new PsFake(false)),
+                0
+            ).enter(new RqFake()).has(),
+            Matchers.is(false)
+        );
+    }
+
+    /**
+     * PsAll with multiple passes that all can be entered returns the
+     * identity specified by an index.
+     * @throws Exception If fails
+     */
+    @Test
+    public void testSuccessfullIdx() throws Exception {
+        final PsFake resulting = new PsFake(true);
+        final RqFake request = new RqFake();
+        MatcherAssert.assertThat(
+            new PsAll(
+                Arrays.asList(
+                    new PsFake(true),
+                    new PsFake(true),
+                    new PsFake(true),
+                    resulting
+                ),
+                Tv.THREE
+            ).enter(request).get().urn(),
+            Matchers.equalTo(resulting.enter(request).get().urn())
```

Comment:
@Suseika are you sure? try replace `Tv.THREE` to `Tv.TWO` I think you will get same result.
I would suggest to use `PsFixed` instead  `PsFake` for `resulting`